### ISSUE TITLE
Wallaby seems to be transforming the same files more than once. Shouldn't it be doing it once and incrementally change any files?

### ISSUE THREAD

**Issue by azaeng04** [wallabyjs/public#3356]
### Issue description or question
Wallaby seems to be transforming the same typescript file more than once when I would expect it to just do it once, cache it and invalidate it after a period of time or based on changes made on the relevant file.

### Wallaby diagnostics report
```
{
  editorVersion: '1.87.2',
  pluginVersion: '1.0.374',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.22631',
  nodeVersion: 'v18.19.0',
  coreVersion: '1.0.1549',
  checksum: 'NDg2ZGU1MzUyMTNiZTU2MDczN2MxZGUyOTEzNWM0NDIsMTcxMTY3MDQwMDAwMCww',
  config: {
    files: [
      { pattern: '**/**/*.ts', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '**/**/*.html', ignore: false, trigger: true, load: true, order: 2 },
      { pattern: 'archive/**/*.*', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'MyApp.Tools.DevopsAutomator.Server/*.*', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'MyApp.Puzzle.Server/*.*', ignore: true, trigger: true, load: true, file: true },
      { pattern: '**/**/*.spec.ts', ignore: true, trigger: true, load: true, file: true },
      { pattern: '**/**/*.test.ts', ignore: true, trigger: true, load: true, file: true },
      { pattern: '.angular/*.*', ignore: true, trigger: true, load: true, file: true },
      { pattern: '**/node_modules/**', ignore: true, trigger: true, load: true, file: true, test: true }
    ],
    tests: [
      { pattern: '**/**/*.spec.ts', ignore: false, trigger: true, load: true, test: true, order: 3 },
      { pattern: '**/**/*.test.ts', ignore: false, trigger: true, load: true, test: true, order: 4 },
      { pattern: '**/node_modules/**', ignore: true, trigger: true, load: true, test: true, file: false }
    ],
    trace: true,
    globalSetupTeardownTimeout: 60000,
    diagnostics: {
      angular: {
        workspace: {
          '$schema': './node_modules/@angular/cli/lib/config/schema.json',
          version: 1,
          projects: {
            'MyApp-cloud': {
              projectType: 'application',
              schematics: {},
              root: '',
              sourceRoot: './',
              prefix: 'app',
              architect: {
                build: {
                  builder: '@angular-devkit/build-angular:browser',
                  options: {
                    tsConfig: './tsconfig.json',
                    index: './MyApp.Cloud.UI/index.html',
                    main: './MyApp.Cloud.UI/boot.ts',
                    assets: [ './assets' ],
                    scripts: [ 'node_modules/marked/marked.min.js' ],
                    styles: [
                      './assets/styles/app.scss',
                      './assets/styles/themes/_core-theme.scss',
                      { input: './assets/styles/themes/default-theme.scss', bundleName: 'default-theme-v2', inject: false },
                      { input: './assets/styles/themes/my-company-theme.scss', bundleName: 'my-company-theme-v2', inject: false },
                      { input: './assets/styles/themes/vital-theme.scss', bundleName: 'vital-theme-v2', inject: false },
                      'node_modules/dock-spawn-ts/lib/css/dock-manager.css',
                      'node_modules/dock-spawn-ts/lib/css/dock-manager-style.css',
                      'node_modules/@fortawesome/fontawesome-free/css/all.min.css',
                      'node_modules/vis-timeline/styles/vis-timeline-graph2d.min.css',
                      'MyApp.WebApps.Slender/flex/assets/css/flex.scss',
                      'MyApp.Cloud.UI/travel-gantt/assets/css/base-styles.scss',
                      'MyApp.Cloud.UI/travel-gantt/assets/css/theme.scss'
                    ],
                    sourceMap: true,
                    outputPath: './MyApp.Cloud.UI.Server/wwwroot'
                  },
                  configurations: {
                    production: {
                      fileReplacements: [ { replace: './environments/working.env.ts', with: './environments/prod.env.ts' } ],
                      outputHashing: 'all',
                      sourceMap: { hidden: true, scripts: true },
                      namedChunks: false,
                      extractLicenses: true,
                      vendorChunk: false,
                      aot: false,
                      buildOptimizer: false,
                      optimization: false,
                      budgets: [ { type: 'initial', maximumWarning: '20mb', maximumError: '25mb' } ]
                    },
                    development: { namedChunks: true, vendorChunk: true, aot: false, buildOptimizer: false, optimization: false, extractLicenses: false }
                  },
                  defaultConfiguration: 'development'
                },
                serve: {
                  builder: '@angular-devkit/build-angular:dev-server',
                  options: { buildTarget: 'MyApp-cloud:build' },
                  configurations: { production: { buildTarget: 'MyApp-cloud:build:production' } }
                },
                'extract-i18n': { builder: '@angular-devkit/build-angular:extract-i18n', options: { buildTarget: 'angular.io-example:build' } },
                lint: { builder: '@angular-eslint/builder:lint', options: { lintFilePatterns: [ '**/**/*.ts', '**/**/*.html' ] } },
                e2e: {
                  builder: '@ngx-playwright/test:run',
                  options: { config: 'MyApp.Cloud.UI.E2E.Testing/playwright.config.ts', baseUrl: 'https://MyAppcloud.MyApplive.com' },
                  configurations: { ci: { forbidOnly: true } }
                }
              }
            }
          },
          schematics: { '@schematics/angular:component': { style: 'scss' } },
          cli: { analytics: false, schematicCollections: [ '@angular-eslint/schematics' ] }
        }
      },
      jest: {
        config: {
          configs: [
            {
              automock: false,
              cache: true,
              cacheDirectory: 'C:\\Users\\MyUser\\AppData\\Local\\Temp\\jest',
              clearMocks: false,
              collectCoverageFrom: [],
              coverageDirectory: 'C:\\Users\\user\\project\\WebApps\\coverage',
              coveragePathIgnorePatterns: [ '\\\\node_modules\\\\' ],
              cwd: 'C:\\Users\\user\\project\\WebApps',
              dependencyExtractor: undefined,
              detectLeaks: false,
              detectOpenHandles: false,
              displayName: undefined,
              errorOnDeprecated: false,
              extensionsToTreatAsEsm: [],
              fakeTimers: { enableGlobally: false },
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: {},
              haste: { computeSha1: false, enableSymlinks: false, forceNodeFilesystemAPI: true, throwOnModuleCollision: false },
              id: '35766cb913839406dfb5bc02c006ef45',
              injectGlobals: true,
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [ 'ts', 'html', 'js', 'json', 'mjs' ],
              moduleNameMapper: [
                [ '^@architecture$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Architecture\\index.ts' ],
                [ '^@logistics$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Logistics\\index.ts' ],
                [ '^@slender/auth$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\auth\\index.ts' ],
                [ '^@slender/testing$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI.Testing\\MyApp.WebApps.Slender\\index.ts' ],
                [ '^@slender/testing/mocks$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI.Testing\\MyApp.WebApps.Slender\\index.ts' ],
                [ '^@slender/flex$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\flex\\index.ts' ],
                [ '^@slender/tools$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\infrastructure\\index.ts' ],
                [ '^@slender/infrastructure$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\infrastructure\\index.ts' ],
                [ '^@slender/foundation$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\foundation\\index.ts' ],
                [ '^@slender/modals$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\modals\\index.ts' ],
                [ '^@slender/input\\-box$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\input-box\\index.ts' ],
                [ '^@slender/list$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\list\\index.ts' ],
                [ '^@slender/expander$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\expander\\index.ts' ],
                [ '^@slender/infinite\\-select$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\infinite-select\\index.ts' ],
                [ '^@slender/date\\-range\\-picker$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\date-range-picker\\index.ts' ],
                [ '^@slender/date\\-range\\-picker2$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\date-range-picker2\\index.ts' ],
                [ '^@slender/ribbon\\-toolbar$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\ribbon-toolbar\\index.ts' ],
                [ '^@slender/data$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\data\\index.ts' ],
                [ '^@slender/maps$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\maps\\index.ts' ],
                [ '^@slender/timeline$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\timeline\\index.ts' ],
                [ '^@slender/calendar$', 'C:\\Users\\user\\project\\WebApps\\MyApp.WebApps.Slender\\calendar\\index.ts' ],
                [ '^@MyApp\\-cloud/platform$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\platform\\index.ts' ],
                [ '^@MyApp\\-cloud/core$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\core\\index.ts' ],
                [ '^@MyApp\\-cloud/graphql$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\core\\graphql\\index.ts' ],
                [ '^@MyApp\\-cloud/data\\-panel$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\data-panel\\index.ts' ],
                [ '^@MyApp\\-cloud/data\\-panel\\-definitions$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\data-panel-definitions\\index.ts' ],
                [ '^@MyApp\\-cloud/forms$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\forms\\index.ts' ],
                [ '^@MyApp\\-cloud/exp\\-forms$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\exp-forms\\index.ts' ],
                [ '^@MyApp\\-cloud/forms\\-infrastructure$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\forms-infrastructure\\index.ts' ],
                [ '^@MyApp\\-cloud/geo\\-coding\\-maps$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\geo-coding-maps\\index.ts' ],
                [ '^@MyApp\\-cloud/timeline\\-gantt$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\timeline-gantt\\index.ts' ],
                [ '^@MyApp\\-cloud/travel\\-maps$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\travel-maps\\index.ts' ],
                [ '^@MyApp\\-cloud/travel\\-gantt$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\travel-gantt\\index.ts' ],
                [ '^@MyApp\\-cloud/system$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\system\\index.ts' ],
                [ '^@MyApp\\-cloud/middleware$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\middleware\\index.ts' ],
                [ '^@MyApp\\-cloud/import\\-export$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\import-export\\index.ts' ],
                [ '^@MyApp\\-cloud/transact$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\transact\\index.ts' ],
                [ '^@MyApp\\-cloud/load\\-builder$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\load-builder\\index.ts' ],
                [ '^@MyApp\\-cloud/calendar$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI\\calendar\\index.ts' ],
                [ '^@MyApp\\-cloud/testing$', 'C:\\Users\\user\\project\\WebApps\\MyApp.Cloud.UI.Testing\\MyApp.Cloud.UI\\index.ts' ],
                [ 'raw-loader!(.*)$', './' ]
              ],
              modulePathIgnorePatterns: [],
              modulePaths: undefined,
              openHandlesTimeout: 1000,
              prettierPath: 'prettier',
              resetMocks: false,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: 'C:\\Users\\user\\project\\WebApps',
              roots: [ 'C:\\Users\\user\\project\\WebApps' ],
              runner: 'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-runner\\build\\index.js',
              runtime: undefined,
              sandboxInjectedGlobals: [],
              setupFiles: [ 'C:\\Users\\user\\project\\WebApps\\jest.polyfill.js' ],
              setupFilesAfterEnv: [ 'C:\\Users\\user\\project\\WebApps\\jest.setup.ts' ],
              skipFilter: false,
              skipNodeResolution: undefined,
              slowTestThreshold: 5,
              snapshotFormat: { escapeString: false, printBasicPrototype: false },
              snapshotResolver: undefined,
              snapshotSerializers: [
                'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-preset-angular\\build\\serializers\\no-ng-attributes.js',
                'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-preset-angular\\build\\serializers\\ng-snapshot.js',
                'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-preset-angular\\build\\serializers\\html-comment.js'
              ],
              testEnvironment: 'C:\\Users\\user\\project\\WebApps\\node_modules\\@happy-dom\\jest-environment\\lib\\index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [ 'C:/Users/MyUser/project/tramm/WebApps/**/__tests__/**/*.[jt]s?(x)', 'C:/Users/MyUser/project/tramm/WebApps/**/*(*.)@(spec|test).[jt]s?(x)' ],
              testPathIgnorePatterns: [ '\\\\node_modules\\\\' ],
              testRegex: [],
              testRunner: 'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-circus\\runner.js',
              transform: [
                [ '^.+\\.js?$', 'C:\\Users\\user\\project\\WebApps\\node_modules\\babel-jest\\build\\index.js', {} ],
                [
                  '^.+\\.(ts|js|mjs|html|svg)$',
                  'C:\\Users\\user\\project\\WebApps\\node_modules\\jest-preset-angular\\build\\index.js',
                  { tsconfig: '<rootDir>/tsconfig.spec.json', stringifyContentPathRegex: '\\.(html|svg)$' }
                ]
              ],
              transformIgnorePatterns: [
                'C:\\\\Users\\\\MyUser\\\\project\\\\tramm\\\\WebApps\\\\node_modules\\\\(?!((.*\\.mjs$)|(@zip.js\\\\zip.js|ol|vis-timeline\\\\esnext|vis-util\\\\esnext|vis-data\\\\esnext|three|uuid)\\\\))'
              ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            ci: false,
            collectCoverage: false,
            collectCoverageFrom: [],
            coverageDirectory: 'C:\\Users\\user\\project\\WebApps\\coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json', 'text', 'lcov', 'clover' ],
            coverageThreshold: undefined,
            detectLeaks: false,
            detectOpenHandles: false,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: false,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 7,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            openHandlesTimeout: 1000,
            outputFile: undefined,
            passWithNoTests: false,
            projects: [],
            randomize: undefined,
            replname: undefined,
            reporters: undefined,
            rootDir: 'C:\\Users\\user\\project\\WebApps',
            runInBand: undefined,
            runTestsByPath: false,
            seed: -920053322,
            shard: undefined,
            showSeed: undefined,
            silent: undefined,
            skipFilter: false,
            snapshotFormat: { escapeString: false, printBasicPrototype: false },
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: 'C:\\Users\\user\\project\\WebApps\\node_modules\\@jest\\test-sequencer\\build\\index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: true,
            watch: false,
            watchAll: false,
            watchPlugins: undefined,
            watchman: true,
            workerIdleMemoryLimit: undefined,
            workerThreads: false
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [],
            watchPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\', 'C:\\Users\\MyUser\\AppData\\Local\\Temp\\jest', '\\.\\coverage' ],
            testPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\', 'C:\\Users\\MyUser\\AppData\\Local\\Temp\\jest', '\\.\\coverage' ],
            testMatch: [ '**/__tests__/**/*.[jt]s?(x)', '**/*(*.)@(spec|test).[jt]s?(x)' ],
            testRegex: []
          }
        }
      }
    },
    filesWithCoverageCalculated: [],
    filesWithNoCoverageCalculated: [],
    globalSetup: false,
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    dot: true,
    captureConsoleLog: true,
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    logLimits: { inline: { depth: 5, elements: 5000 }, values: { default: { stringLength: 8192 }, autoExpand: { elements: 5000, stringLength: 8192, depth: 10 } } },
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore|c8 ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    env: { type: 'node', params: {}, runner: 'C:\\Program Files\\nodejs\\node.exe', viewportSize: { width: 800, height: 600 }, options: { width: 800, height: 600 }, bundle: true },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    runAllTestsWhenNoAffectedTests: false,
    configCode: '\r\n' +
      'module.exports = function (){\r\n' +
      '  return {\r\n' +
      '    autoDetect: true,\r\n' +
      '    files: [\r\n' +
      '      "**/**/*.ts",\r\n' +
      '      "**/**/*.html",\r\n' +
      '      "!archive/**/*.*",\r\n' +
      '      "!MyApp.Tools.DevopsAutomator.Server/*.*",\r\n' +
      '      "!MyApp.Puzzle.Server/*.*",\r\n' +
      '      "!**/**/*.spec.ts",\r\n' +
      '      "!**/**/*.test.ts",\r\n' +
      '      "!.angular/*.*",\r\n' +
      '      "!**/node_modules/**"\r\n' +
      '    ],\r\n' +
      '    tests: ["**/**/*.spec.ts", "**/**/*.test.ts", "!**/node_modules/**"],\r\n' +
      '    trace: true,\r\n' +
      '    globalSetupTeardownTimeout: 60 * 1000\r\n' +
      '  };\r\n' +
      '};\r\n'
  },
  packageJSON: {
    dependencies: {
      '@angular/core': '17.1.1',
      '@angular/common': '17.1.1',
      '@angular/compiler': '17.1.1',
      '@angular/platform-browser': '17.1.1',
      '@angular/platform-browser-dynamic': '17.1.1',
      '@angular/router': '17.1.1',
      '@angular/forms': '17.1.1',
      '@angular/animations': '17.1.1',
      '@angular/localize': '17.1.1',
      '@angular/cdk': '17.1.1',
      '@angular/material': '17.1.1',
      '@ngneat/until-destroy': '9.2.0',
      '@boldbi/boldbi-embedded-sdk': '6.3.3',
      '@microsoft/signalr': '7.0.3',
      '@azure/msal-browser': '3.1.0',
      '@azure/msal-angular': '3.0.4',
      '@fullcalendar/angular': '^6.1.10',
      '@fullcalendar/core': '^6.1.10',
      '@fullcalendar/daygrid': '^6.1.10',
      '@fullcalendar/interaction': '^6.1.10',
      'zone.js': '0.14.3',
      rxjs: '7.8.0',
      tslib: '2.6.2',
      util: '0.12.4',
      'ngx-resize-observer': '2.6.0',
      'text-encoding-shim': '1.0.4',
      'ngx-cookie-service': '17.0.1',
      'fast-sha256': '1.3.0',
      'date-fns': '2.16.1',
      'date-fns-tz': '1.1.6',
      'dock-spawn-ts': '2.403.x',
      lodash: '4.17.21',
      pluralize: '8.0.0',
      papaparse: '5.4.1',
      'file-saver': '2.0.2',
      'leader-line-new': '1.1.9',
      ol: '6.14.x',
      '@fortawesome/fontawesome-free': '6.5.1',
      'vis-data': '7.1.6',
      'vis-network': '9.1.6',
      'vis-timeline': '7.7.2',
      'vis-util': '5.0.3',
      xlsx: '0.18.5',
      '@techiediaries/ngx-qrcode': '9.1.0',
      '@zip.js/zip.js': '2.3.24',
      'xml-js': '1.6.11',
      stream: '0.0.2',
      'ngx-markdown': '17.1.1',
      'dat.gui': '0.7.7',
      three: '0.148.0',
      'ngx-monaco-editor-v2': '17.0.1',
      'object-hash': '3.0.0'
    },
    devDependencies: {
      '@angular/cli': '17.1.1',
      '@angular/compiler-cli': '17.1.1',
      '@angular/language-service': '17.1.1',
      '@angular-devkit/build-angular': '17.1.1',
      '@angular-eslint/builder': '17.1.1',
      '@angular-eslint/eslint-plugin': '17.1.1',
      '@angular-eslint/eslint-plugin-template': '17.1.1',
      '@angular-eslint/schematics': '17.1.1',
      '@angular-eslint/template-parser': '17.1.1',
      '@typescript-eslint/eslint-plugin': '6.10.0',
      '@typescript-eslint/parser': '6.10.0',
      typescript: '5.2.2',
      eslint: '8.53.0',
      'raw-loader': '4.0.2',
      'url-loader': '4.1.1',
      graphql: '14.5.8',
      '@jridgewell/trace-mapping': '0.3.20',
      '@graphql-codegen/cli': '1.8.3',
      '@graphql-codegen/add': '1.8.3',
      '@graphql-codegen/typescript': '1.8.3',
      '@graphql-codegen/typescript-operations': '1.8.3',
      jest: '29.7.0',
      'ts-jest': '29.1.2',
      'jest-preset-angular': '14.0.0',
      'jest-auto-spies': '3.0.0',
      '@happy-dom/jest-environment': '14.0.0',
      'ts-node': '10.9.1',
      msw: '2.2.12',
      marked: '9.1.6',
      '@babel/core': '7.24.3',
      '@types/papaparse': '5.3.14',
      '@types/file-saver': '2.0.1',
      '@types/node': '18.17.14',
      '@types/ws': '8.5.4',
      '@types/pluralize': '0.0.29',
      '@types/ol': '6.5.x',
      '@types/pdfmake': '0.1.17',
      '@types/lodash': '4.14.191',
      '@types/dat.gui': '0.7.7',
      '@types/three': '0.148.1',
      '@types/object-hash': '3.0.4',
      '@types/marked': '4.3.0',
      '@types/jest': '29.5.11',
      '@ngx-playwright/test': '^0.5.3',
      '@babel/preset-env': '7.24.3',
      'babel-plugin-transform-import-meta': '2.2.1',
      'babel-jest': '29.7.0',
      moment: '2.30.1',
      undici: '6.11.1',
      '@babel/preset-typescript': '7.24.1',
      '@babel/plugin-proposal-decorators': '7.24.1'
    }
  },
  fs: { numberOfFiles: 2713 },
  debug: [
    '2024-04-04T11:21:06.765Z project waiting for initial run signal\n',
    '2024-04-04T11:21:06.866Z config Attempting automatic configuration for angular\n',
    '2024-04-04T11:21:06.886Z angular/cli config Detected Angular CLI.\n',
    '2024-04-04T11:21:06.900Z angular/cli config Angular currentDirPath: "C:\\Users\\user\\project\\WebApps", workspaceDirPath: "C:\\Users\\user\\project\\WebApps"\n',
    '2024-04-04T11:21:06.900Z angular/cli config Error: None of your projects are configured with `@angular-devkit/build-angular:karma`, `@angular-builders/custom-webpack:karma`, `@angular-builders/jest:run`, `@nrwl/jest:jest` or `@nx/jest:jest` builders as their test architect.\n' +
      '    at t.configure (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:156:17985)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12595)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12480)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12480)\n' +
      '    at async Config.load (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12420)\n',
    '2024-04-04T11:21:06.902Z angular/cli config Angular.json: \n' +
      ' {\n' +
      "  '$schema': './node_modules/@angular/cli/lib/config/schema.json',\n" +
      '  version: 1,\n' +
      '  projects: {\n' +
      "    'MyApp-cloud': {\n" +
      "      projectType: 'application',\n" +
      '      schematics: {},\n' +
      "      root: '',\n" +
      "      sourceRoot: './',\n" +
      "      prefix: 'app',\n" +
      '      architect: {\n' +
      '        build: {\n' +
      "          builder: '@angular-devkit/build-angular:browser',\n" +
      '          options: {\n' +
      "            tsConfig: './tsconfig.json',\n" +
      "            index: './MyApp.Cloud.UI/index.html',\n" +
      "            main: './MyApp.Cloud.UI/boot.ts',\n" +
      "            assets: [ './assets' ],\n" +
      "            scripts: [ 'node_modules/marked/marked.min.js' ],\n" +
      '            styles: [\n' +
      "              './assets/styles/app.scss',\n" +
      "              './assets/styles/themes/_core-theme.scss',\n" +
      "              { input: './assets/styles/themes/default-theme.scss', bundleName: 'default-theme-v2', inject: false },\n" +
      "              { input: './assets/styles/themes/my-company-theme.scss', bundleName: 'my-company-theme-v2', inject: false },\n" +
      "              { input: './assets/styles/themes/vital-theme.scss', bundleName: 'vital-theme-v2', inject: false },\n" +
      "              'node_modules/dock-spawn-ts/lib/css/dock-manager.css',\n" +
      "              'node_modules/dock-spawn-ts/lib/css/dock-manager-style.css',\n" +
      "              'node_modules/@fortawesome/fontawesome-free/css/all.min.css',\n" +
      "              'node_modules/vis-timeline/styles/vis-timeline-graph2d.min.css',\n" +
      "              'MyApp.WebApps.Slender/flex/assets/css/flex.scss',\n" +
      "              'MyApp.Cloud.UI/travel-gantt/assets/css/base-styles.scss',\n" +
      "              'MyApp.Cloud.UI/travel-gantt/assets/css/theme.scss'\n" +
      '            ],\n' +
      '            sourceMap: true,\n' +
      "            outputPath: './MyApp.Cloud.UI.Server/wwwroot'\n" +
      '          },\n' +
      '          configurations: {\n' +
      '            production: {\n' +
      "              fileReplacements: [ { replace: './environments/working.env.ts', with: './environments/prod.env.ts' } ],\n" +
      "              outputHashing: 'all',\n" +
      '              sourceMap: { hidden: true, scripts: true },\n' +
      '              namedChunks: false,\n' +
      '              extractLicenses: true,\n' +
      '              vendorChunk: false,\n' +
      '              aot: false,\n' +
      '              buildOptimizer: false,\n' +
      '              optimization: false,\n' +
      "              budgets: [ { type: 'initial', maximumWarning: '20mb', maximumError: '25mb' } ]\n" +
      '            },\n' +
      '            development: { namedChunks: true, vendorChunk: true, aot: false, buildOptimizer: false, optimization: false, extractLicenses: false }\n' +
      '          },\n' +
      "          defaultConfiguration: 'development'\n" +
      '        },\n' +
      "        serve: { builder: '@angular-devkit/build-angular:dev-server', options: { buildTarget: 'MyApp-cloud:build' }, configurations: { production: { buildTarget: 'MyApp-cloud:build:production' } } },\n" +
      "        'extract-i18n': { builder: '@angular-devkit/build-angular:extract-i18n', options: { buildTarget: 'angular.io-example:build' } },\n" +
      "        lint: { builder: '@angular-eslint/builder:lint', options: { lintFilePatterns: [ '**/**/*.ts', '**/**/*.html' ] } },\n" +
      '        e2e: {\n' +
      "          builder: '@ngx-playwright/test:run',\n" +
      "          options: { config: 'MyApp.Cloud.UI.E2E.Testing/playwright.config.ts', baseUrl: 'https://MyAppcloud.MyApplive.com' },\n" +
      '          configurations: { ci: { forbidOnly: true } }\n' +
      '        }\n' +
      '      }\n' +
      '    }\n' +
      '  },\n' +
      "  schematics: { '@schematics/angular:component': { style: 'scss' } },\n" +
      "  cli: { analytics: false, schematicCollections: [ '@angular-eslint/schematics' ] }\n" +
      '}\n',
    '2024-04-04T11:21:06.902Z angular/cli config Error: None of your projects are configured with `@angular-devkit/build-angular:karma`, `@angular-builders/custom-webpack:karma`, `@angular-builders/jest:run`, `@nrwl/jest:jest` or `@nx/jest:jest` builders as their test architect.\n' +
      '    at t.configure (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:156:17985)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12595)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12480)\n' +
      '    at async automaticConfigurationProviders.reduce.Promise.resolve.success (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12480)\n' +
      '    at async Config.load (c:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\wallaby3d785f\\server.js:144:12420)\n',
    '2024-04-04T11:21:06.902Z config Finished attempting automatic configuration for angular (36ms)\n',
    '2024-04-04T11:21:06.902Z config Attempting automatic configuration for jest\n',
    '2024-04-04T11:21:10.365Z jest/config Detected Jest.\n',
    '2024-04-04T11:21:10.366Z jest/config Configured Jest.\n',
    '2024-04-04T11:21:10.367Z config Finished attempting automatic configuration for jest (3464ms)\n',
    '2024-04-04T11:21:10.369Z project Wallaby Node version: v18.19.0\n',
    '2024-04-04T11:21:10.369Z project Wallaby config: C:\\Users\\user\\project\\WebApps\\wallaby.js\n',
    '2024-04-04T11:21:10.410Z fs File system starting\n',
    '2024-04-04T11:21:14.655Z fs File system scan completed\n',
    '2024-04-04T11:21:14.673Z project File cache: C:\\Users\\MyUser\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.374\\projects\\0f01cce986cb2c0e\n',
    '2024-04-04T11:21:16.987Z uiService Listening port 51235\n',
    '2024-04-04T11:21:17.023Z project package.json file change detected, invalidating local cache\n',
    '2024-04-04T11:21:17.090Z workers Parallelism for initial run: 6, for regular run: 3\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #0\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #1\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #2\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #3\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #4\n',
    '2024-04-04T11:21:17.090Z workers Starting run worker instance #5\n',
    '2024-04-04T11:21:17.091Z workers Web server is listening at 54838\n',
    '2024-04-04T11:21:17.091Z project File cache requires some updates, waiting required files from IDE\n',
    '2024-04-04T11:21:17.227Z project Stopping process pool\n',
    '2024-04-04T11:21:17.233Z project Test run started; run priority: 3\n',
    '2024-04-04T11:21:17.245Z project Running all tests\n',
    '2024-04-04T11:21:17.325Z workers Starting test run, priority: 3\n',
    '2024-04-04T11:21:17.326Z workers Distributing tests between 6 workers\n',
    '2024-04-04T11:21:17.331Z workers Running tests in parallel\n',
    '2024-04-04T11:21:17.332Z nodeRunner Starting sandbox [worker #0, session #7caqk]\n',
    '2024-04-04T11:21:17.332Z nodeRunner Starting sandbox [worker #1, session #acz9l]\n',
    '2024-04-04T11:21:17.332Z nodeRunner Starting sandbox [worker #2, session #v3ntr]\n',
    '2024-04-04T11:21:17.333Z nodeRunner Preparing sandbox [worker #0, session #7caqk]\n',
    '2024-04-04T11:21:17.333Z nodeRunner Preparing sandbox [worker #1, session #acz9l]\n',
    '2024-04-04T11:21:17.333Z nodeRunner Preparing sandbox [worker #2, session #v3ntr]\n',
    '2024-04-04T11:21:17.730Z workers Started run worker instance (delayed) #3\n',
    '2024-04-04T11:21:17.732Z workers Started run worker instance (delayed) #0\n',
    '2024-04-04T11:21:17.732Z nodeRunner Prepared sandbox [worker #0, session #7caqk]\n',
    '2024-04-04T11:21:17.733Z workers [worker #0, session #7caqk] Running tests in sandbox\n',
    '2024-04-04T11:21:17.742Z workers Started run worker instance (delayed) #5\n',
    '2024-04-04T11:21:17.743Z workers Started run worker instance (delayed) #2\n',
    '2024-04-04T11:21:17.743Z nodeRunner Prepared sandbox [worker #2, session #v3ntr]\n',
    '2024-04-04T11:21:17.743Z workers [worker #2, session #v3ntr] Running tests in sandbox\n',
    '2024-04-04T11:21:17.754Z workers Started run worker instance (delayed) #4\n',
    '2024-04-04T11:21:17.754Z workers Started run worker instance (delayed) #1\n',
    '2024-04-04T11:21:17.755Z nodeRunner Prepared sandbox [worker #1, session #acz9l]\n',
    '2024-04-04T11:21:17.755Z workers [worker #1, session #acz9l] Running tests in sandbox\n',
    "2024-04-04T11:21:23.518Z workers 'Scheduling Jest Test Run (acz9l): 2024-04-04T11:21:23.516Z'\n",
    "2024-04-04T11:21:23.518Z workers 'Scheduling Jest Test Run (v3ntr): 2024-04-04T11:21:23.516Z'\n",
    "2024-04-04T11:21:23.691Z workers 'Scheduling Jest Test Run (7caqk): 2024-04-04T11:21:23.688Z'\n",
    '2024-04-04T11:21:40.400Z fs File changed in editor: jest.setup.ts\n',
    '2024-04-04T11:21:40.401Z project Test run is not cancelled because cancel requester does not have enough priority to cancel the run\n',
    '2024-04-04T11:21:41.671Z fs File changed: jest.setup.ts\n',
    '2024-04-04T11:21:41.721Z fs No changes detected for jest.setup.ts\n',
    '2024-04-04T11:21:41.721Z extended-core File was not changed, but file markers may need to be synced\n',
    '2024-04-04T11:23:11.519Z workers [acz9l] Loaded unknown number of test(s)\n',
    '2024-04-04T11:23:12.565Z workers [acz9l] Test executed: should be defined\n',
    '2024-04-04T11:23:13.514Z workers [acz9l] Test executed: should select start and end date\n',
    '2024-04-04T11:23:13.872Z workers [acz9l] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:23:13.873Z workers 'Jest Test Run Complete (acz9l): 2024-04-04T11:23:13.872Z'\n",
    '2024-04-04T11:23:13.882Z workers [acz9l] Sandbox is responsive, closing it\n',
    '2024-04-04T11:23:15.073Z workers [7caqk] Loaded unknown number of test(s)\n',
    '2024-04-04T11:23:15.937Z workers [7caqk] Test executed: should disable Continue button disabled when logging in\n',
    '2024-04-04T11:23:15.940Z workers Failed to map the stack to user code, entry message: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button"), stack: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button")\n' +
      '    at C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:599:15\n' +
      '    at Generator.next (<anonymous>)\n' +
      '    at fulfilled (C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:22:24)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone.umd.js:411:30)\n' +
      '    at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone-testing.umd.js:300:43)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone\n',
    '2024-04-04T11:23:16.111Z workers [7caqk] Test executed: should disable Continue button when accepting Ts&Cs\n',
    '2024-04-04T11:23:16.112Z workers Failed to map the stack to user code, entry message: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button"), stack: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button")\n' +
      '    at C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:599:15\n' +
      '    at Generator.next (<anonymous>)\n' +
      '    at fulfilled (C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:22:24)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone.umd.js:411:30)\n' +
      '    at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone-testing.umd.js:300:43)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone\n',
    '2024-04-04T11:23:16.248Z workers [7caqk] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:23:16.248Z workers 'Jest Test Run Complete (7caqk): 2024-04-04T11:23:16.248Z'\n",
    '2024-04-04T11:23:16.262Z workers [7caqk] Sandbox is responsive, closing it\n',
    '2024-04-04T11:25:37.262Z workers [v3ntr] Loaded unknown number of test(s)\n',
    '2024-04-04T11:25:40.178Z workers [v3ntr] Test executed: should be defined\n',
    '2024-04-04T11:25:40.240Z workers [v3ntr] Test executed: should save date changes\n',
    '2024-04-04T11:25:40.770Z workers [v3ntr] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:25:40.771Z workers 'Jest Test Run Complete (v3ntr): 2024-04-04T11:25:40.770Z'\n",
    '2024-04-04T11:25:40.784Z workers [v3ntr] Sandbox is responsive, closing it\n',
    '2024-04-04T11:25:40.784Z workers Merging parallel test run results\n',
    '2024-04-04T11:25:40.802Z project Test run finished\n',
    '2024-04-04T11:25:40.803Z project Processed console.log entries\n',
    '2024-04-04T11:25:40.818Z project Processed loading sequences\n',
    '2024-04-04T11:25:40.821Z project Processed executed tests\n',
    '2024-04-04T11:25:41.239Z project Processed code coverage\n',
    '2024-04-04T11:25:42.071Z project Test run result processed and sent to IDE\n',
    '2024-04-04T11:25:42.108Z project Test run started; run priority: 2\n',
    '2024-04-04T11:25:42.110Z testTask Test files from affected: 3, from deleted or manually requested: 0, from recently changed: 0, from loaded by: 0, from failing: 2\n',
    '2024-04-04T11:25:42.172Z workers Starting test run, priority: 2\n',
    '2024-04-04T11:25:42.172Z workers Distributing tests between 3 workers\n',
    '2024-04-04T11:25:42.173Z workers Running tests in parallel\n',
    '2024-04-04T11:25:42.173Z nodeRunner Starting sandbox [worker #0, session #t03k4]\n',
    '2024-04-04T11:25:42.173Z nodeRunner Starting sandbox [worker #1, session #18usy]\n',
    '2024-04-04T11:25:42.173Z nodeRunner Starting sandbox [worker #2, session #ehq15]\n',
    '2024-04-04T11:25:42.173Z nodeRunner Preparing sandbox [worker #0, session #t03k4]\n',
    '2024-04-04T11:25:42.173Z nodeRunner Preparing sandbox [worker #1, session #18usy]\n',
    '2024-04-04T11:25:42.173Z nodeRunner Preparing sandbox [worker #2, session #ehq15]\n',
    '2024-04-04T11:25:42.174Z nodeRunner Prepared sandbox [worker #0, session #t03k4]\n',
    '2024-04-04T11:25:42.174Z nodeRunner Prepared sandbox [worker #1, session #18usy]\n',
    '2024-04-04T11:25:42.174Z nodeRunner Prepared sandbox [worker #2, session #ehq15]\n',
    '2024-04-04T11:25:42.174Z workers [worker #0, session #t03k4] Running tests in sandbox\n',
    '2024-04-04T11:25:42.190Z workers [worker #1, session #18usy] Running tests in sandbox\n',
    '2024-04-04T11:25:42.253Z workers [worker #2, session #ehq15] Running tests in sandbox\n',
    "2024-04-04T11:25:42.406Z workers 'Scheduling Jest Test Run (t03k4): 2024-04-04T11:25:42.329Z'\n",
    "2024-04-04T11:25:42.407Z workers 'Scheduling Jest Test Run (ehq15): 2024-04-04T11:25:42.351Z'\n",
    "2024-04-04T11:25:42.407Z workers 'Scheduling Jest Test Run (18usy): 2024-04-04T11:25:42.372Z'\n",
    '2024-04-04T11:25:47.282Z workers [t03k4] Loaded unknown number of test(s)\n',
    '2024-04-04T11:25:47.324Z workers [18usy] Loaded unknown number of test(s)\n',
    '2024-04-04T11:25:47.985Z workers [18usy] Test executed: should be defined\n',
    '2024-04-04T11:25:48.131Z workers [t03k4] Test executed: should disable Continue button disabled when logging in\n',
    '2024-04-04T11:25:48.131Z workers Failed to map the stack to user code, entry message: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button"), stack: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button")\n' +
      '    at C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:599:15\n' +
      '    at Generator.next (<anonymous>)\n' +
      '    at fulfilled (C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:22:24)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone.umd.js:411:30)\n' +
      '    at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone-testing.umd.js:300:43)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone\n',
    '2024-04-04T11:25:48.359Z workers [t03k4] Test executed: should disable Continue button when accepting Ts&Cs\n',
    '2024-04-04T11:25:48.359Z workers Failed to map the stack to user code, entry message: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button"), stack: Error: Failed to find element matching one of the following queries:\n' +
      '(_MatButtonHarness with host element matching selector: "[mat-button], [mat-raised-button], [mat-flat-button],\n' +
      '                         [mat-icon-button], [mat-stroked-button], [mat-fab], [mat-mini-fab]" satisfying the constraints: host matches selector "#login-continue-button")\n' +
      '    at C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:599:15\n' +
      '    at Generator.next (<anonymous>)\n' +
      '    at fulfilled (C:\\Users\\user\\project\\WebApps\\node_modules\\@angular\\cdk\\fesm2022\\testing.mjs:22:24)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone.umd.js:411:30)\n' +
      '    at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone.js\\bundles\\zone-testing.umd.js:300:43)\n' +
      '    at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (C:\\Users\\user\\project\\WebApps\\node_modules\\zone\n',
    '2024-04-04T11:25:48.647Z workers [18usy] Test executed: should select start and end date\n',
    '2024-04-04T11:25:48.697Z workers [t03k4] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:25:48.698Z workers 'Jest Test Run Complete (t03k4): 2024-04-04T11:25:48.697Z'\n",
    '2024-04-04T11:25:48.705Z workers [t03k4] Sandbox is responsive, closing it\n',
    '2024-04-04T11:25:49.055Z workers [18usy] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:25:49.056Z workers 'Jest Test Run Complete (18usy): 2024-04-04T11:25:49.055Z'\n",
    '2024-04-04T11:25:49.057Z workers [18usy] Sandbox is responsive, closing it\n',
    '2024-04-04T11:25:50.772Z workers [ehq15] Loaded unknown number of test(s)\n',
    '2024-04-04T11:25:53.068Z workers [ehq15] Test executed: should be defined\n',
    '2024-04-04T11:25:53.152Z workers [ehq15] Test executed: should save date changes\n',
    '2024-04-04T11:25:53.529Z workers [ehq15] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:25:53.529Z workers 'Jest Test Run Complete (ehq15): 2024-04-04T11:25:53.528Z'\n",
    '2024-04-04T11:25:53.529Z workers [ehq15] Sandbox is responsive, closing it\n',
    '2024-04-04T11:25:53.529Z workers Merging parallel test run results\n',
    '2024-04-04T11:25:53.584Z project Test run finished\n',
    '2024-04-04T11:25:53.585Z project Processed console.log entries\n',
    '2024-04-04T11:25:53.589Z project Processed loading sequences\n',
    '2024-04-04T11:25:53.589Z project Processed executed tests\n',
    '2024-04-04T11:25:53.838Z project Processed code coverage\n',
    '2024-04-04T11:25:54.025Z project Test run result processed and sent to IDE\n',
    '2024-04-04T11:29:47.107Z project Requested to run some tests\n',
    '2024-04-04T11:29:47.124Z project Test run started; run priority: 1\n',
    '2024-04-04T11:29:47.124Z testTask Test files from affected: 0, from deleted or manually requested: 1, from recently changed: 0, from loaded by: 0, from failing: 0\n',
    '2024-04-04T11:29:47.164Z workers Starting test run, priority: 1\n',
    '2024-04-04T11:29:47.164Z nodeRunner Starting sandbox [worker #0, session #938tb]\n',
    '2024-04-04T11:29:47.164Z nodeRunner Preparing sandbox [worker #0, session #938tb]\n',
    '2024-04-04T11:29:47.164Z nodeRunner Prepared sandbox [worker #0, session #938tb]\n',
    '2024-04-04T11:29:47.164Z workers [worker #0, session #938tb] Running tests in sandbox\n',
    "2024-04-04T11:29:47.179Z workers 'Scheduling Jest Test Run (938tb): 2024-04-04T11:29:47.179Z'\n",
    '2024-04-04T11:29:58.225Z fs File changed in editor: MyApp.Cloud.UI.Testing/MyApp.Cloud.UI/system/components/tests/header-control.component.ui.test.ts\n',
    '2024-04-04T11:29:58.229Z workers Cancelling test run, cancel requester priority: 2, current run priority: 1\n',
    '2024-04-04T11:29:58.229Z project Test run was cancelled\n',
    '2024-04-04T11:29:59.126Z fs File changed: MyApp.Cloud.UI.Testing/MyApp.Cloud.UI/system/components/tests/header-control.component.ui.test.ts\n',
    '2024-04-04T11:29:59.185Z fs No changes detected for MyApp.Cloud.UI.Testing/MyApp.Cloud.UI/system/components/tests/header-control.component.ui.test.ts\n',
    '2024-04-04T11:29:59.185Z extended-core File was not changed, but file markers may need to be synced\n',
    '2024-04-04T11:30:05.228Z workers [938tb] Sandbox is not responsive, recycling worker instance\n',
    '2024-04-04T11:30:05.229Z workers Recyling run worker instance #0\n',
    '2024-04-04T11:30:05.230Z project Test run cancelled, re-queueing run data\n',
    '2024-04-04T11:30:05.230Z project Test run finished\n',
    '2024-04-04T11:30:05.231Z project Test run data re-queued\n',
    '2024-04-04T11:30:05.305Z project Test run started; run priority: 2\n',
    '2024-04-04T11:30:05.306Z testTask Test files from affected: 1, from deleted or manually requested: 0, from recently changed: 0, from loaded by: 0, from failing: 0\n',
    '2024-04-04T11:30:05.427Z workers Starting test run, priority: 2\n',
    '2024-04-04T11:30:05.427Z nodeRunner Starting sandbox [worker #0, session #loaka]\n',
    '2024-04-04T11:30:05.427Z nodeRunner Preparing sandbox [worker #0, session #loaka]\n',
    '2024-04-04T11:30:05.428Z workers Starting run worker instance #0\n',
    '2024-04-04T11:30:07.552Z workers Started run worker instance (delayed) #0\n',
    '2024-04-04T11:30:07.552Z nodeRunner Prepared sandbox [worker #0, session #loaka]\n',
    '2024-04-04T11:30:07.552Z workers [worker #0, session #loaka] Running tests in sandbox\n',
    "2024-04-04T11:30:28.374Z workers 'Scheduling Jest Test Run (loaka): 2024-04-04T11:30:28.373Z'\n",
    '2024-04-04T11:31:54.731Z workers Some long running code has been detected: one of your files is taking more than 5000ms to execute.\n' +
      'Execution of the following files has started but has not finished:\n' +
      '- MyApp.WebApps.Logistics/index.ts\n' +
      '- MyApp.Cloud.UI/core/index.ts\n' +
      '- MyApp.Cloud.UI/platform/index.ts\n' +
      '- MyApp.WebApps.Slender/auth/index.ts\n' +
      '- MyApp.WebApps.Logistics/types/date-range.ts\n' +
      '- MyApp.WebApps.Logistics/types/index.ts\n' +
      '- MyApp.WebApps.Slender/foundation/boot.module.ts\n' +
      '- MyApp.WebApps.Slender/foundation/index.ts\n' +
      '- MyApp.Cloud.UI/core/controllers/data-sync.controller.ts\n' +
      '- MyApp.Cloud.UI/core/controllers/index.ts\n' +
      '- MyApp.Cloud.UI/core/components/index.ts\n' +
      '- MyApp.Cloud.UI/core/components/ref.component.ts\n' +
      '- MyApp.Cloud.UI/platform/components/index.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/index.ts\n' +
      '- MyApp.Cloud.UI/core/services/providers/platform-config.provider.ts\n' +
      '- MyApp.Cloud.UI/platform/components/_layout/layout.component.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/admin-panel.component.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/admin.module.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/index.ts\n' +
      '- MyApp.Cloud.UI.Testing/MyApp.Cloud.UI/system/components/tests/header-control.component.ui.test.ts\n' +
      'Try commenting out the test or excluding the test file from the `tests` list in your wallaby config,\n' +
      'and restarting wallaby to make sure that it is this test/file causing the issue and not something else.\n' +
      'Pinging test runner sandbox...\n',
    '2024-04-04T11:31:54.934Z workers The sandbox is not responsive. Check for possibly recently introduced infinite loops.\n',
    '2024-04-04T11:32:23.693Z workers Some long running code has been detected: one of your files is taking more than 5000ms to execute.\n' +
      'Execution of the following files has started but has not finished:\n' +
      '- MyApp.Cloud.UI/core/index.ts\n' +
      '- MyApp.Cloud.UI/platform/index.ts\n' +
      '- MyApp.WebApps.Slender/auth/index.ts\n' +
      '- MyApp.WebApps.Slender/maps/index.ts\n' +
      '- MyApp.WebApps.Slender/maps/package.module.ts\n' +
      '- MyApp.WebApps.Slender/foundation/boot.module.ts\n' +
      '- MyApp.WebApps.Slender/foundation/index.ts\n' +
      '- MyApp.Cloud.UI/core/controllers/data-sync.controller.ts\n' +
      '- MyApp.Cloud.UI/core/controllers/index.ts\n' +
      '- MyApp.Cloud.UI/core/components/index.ts\n' +
      '- MyApp.Cloud.UI/core/components/ref.component.ts\n' +
      '- MyApp.Cloud.UI/core/infrastructure/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/index.ts\n' +
      '- MyApp.Cloud.UI/platform/components/index.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/index.ts\n' +
      '- MyApp.WebApps.Slender/maps/components/index.ts\n' +
      '- MyApp.WebApps.Slender/maps/components/main.component.ts\n' +
      '- MyApp.Cloud.UI/core/infrastructure/global-search/entity-search-context.ts\n' +
      '- MyApp.Cloud.UI/core/infrastructure/global-search/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/master/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/master/region.ts\n' +
      '- MyApp.Cloud.UI/core/model/master/site.ts\n' +
      '- MyApp.Cloud.UI/core/infrastructure/helpers/add-custom-state-machine-props.helper.ts\n' +
      '- MyApp.Cloud.UI/core/infrastructure/helpers/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/_system/index.ts\n' +
      '- MyApp.Cloud.UI/core/services/providers/platform-config.provider.ts\n' +
      '- MyApp.Cloud.UI/platform/components/_layout/layout.component.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/admin-panel.component.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/admin.module.ts\n' +
      '- MyApp.WebApps.Slender/auth/features/admin/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/_system/complex-types/crossdock-cutoff-time-schedule.ts\n' +
      '- MyApp.Cloud.UI/core/model/_system/complex-types/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/master/complex-types/index.ts\n' +
      '- MyApp.Cloud.UI/core/model/master/complex-types/lane-segment.ts\n' +
      '- MyApp.Cloud.UI.Testing/MyApp.Cloud.UI/system/components/tests/header-control.component.ui.test.ts\n' +
      'Try commenting out the test or excluding the test file from the `tests` list in your wallaby config,\n' +
      'and restarting wallaby to make sure that it is this test/file causing the issue and not something else.\n' +
      'Pinging test runner sandbox...\n',
    '2024-04-04T11:33:53.471Z workers [loaka] Loaded unknown number of test(s)\n',
    '2024-04-04T11:33:55.729Z workers [loaka] Test executed: should be defined\n',
    '2024-04-04T11:33:55.810Z workers [loaka] Test executed: should save date changes\n',
    '2024-04-04T11:33:55.953Z workers Sandbox is responsive. The issue may be asynchronous in nature (for example, a missing callback).\n',
    '2024-04-04T11:33:55.954Z workers Sandbox is responsive. The issue may be asynchronous in nature (for example, a missing callback).\n',
    '2024-04-04T11:33:56.259Z workers [loaka] Run 2 test(s), skipped 0 test(s)\n',
    "2024-04-04T11:33:56.260Z workers 'Jest Test Run Complete (loaka): 2024-04-04T11:33:56.260Z'\n",
    '2024-04-04T11:33:56.265Z workers [loaka] Sandbox is responsive, closing it\n',
    '2024-04-04T11:33:56.266Z project Test run finished\n',
    '2024-04-04T11:33:56.266Z project Processed console.log entries\n',
    '2024-04-04T11:33:56.267Z project Processed loading sequences\n',
    '2024-04-04T11:33:56.268Z project Processed executed tests\n',
    '2024-04-04T11:33:56.451Z project Processed code coverage\n',
    '2024-04-04T11:33:56.641Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by smcenlly** [comment]
Closing this because we need more detail to determine whether Wallaby is doing duplicate work or whether the observed behavior is expected for the project setup, and there has not been a follow-up since we asked for that information.

If this is still a problem, please open a new issue with the signs you are seeing, the diagnostics report, and ideally a small sample project.
